### ISSUE TITLE
XmlSerializer.GenerateSerializer should not be in System.Private.Xml

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#1413]
System.Private.Xml contains all of the code for generating an assembly as the core logic behind the sgen tool:
https://github.com/dotnet/corefx/tree/master/src/Microsoft.XmlSerializer.Generator
The vast majority of the functionality here isn't used by anything other than sgen, which calls XmlSerializer.GenerateSerializer via reflection, and is the only caller of that method.  This functionality should be removed from XmlSerializer in System.Private.Xml and moved into sgen.

**Comment by StephenBonikowsky** [comment]
@mconnew Please comment on this issue and resolve as appropriate.

**Comment by eerhardt** [comment]
> I maintain code that uses System.Xml.Serialization.XmlSerializer.GenerateSerializer (directly, not using reflection)

@tamlin-mike - can you explain this more? Are you only doing that on the .NET Framework? This method is `internal` in .NET Core.

https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs#L596

**Comment by tamlin-mike** [comment]
> This method is internal in .NET Core.

Yeah, that turned out to be the problem. It was public in Framework. Unfortunately the code started to depend on that functionality. Then the need arose to use some other assembly compiled to netstandard 2.0 IIRC, and much gnashing of teeth ensued. Ended up throwing the MS tool out and writing our own serialization generator (-frontend) to get the required cross-platform compatibility we needed (fw + netstd). Unfortunately that now depends on that older specific version of the mentioned nuget package -- that will likely be removed sooner or later, or become incompatible with later versions of some runtime -- so probability is high we'll either be forced to rewrite it all from scratch or simply burn that use of XML at the stake. To quote Alien "Nuke it from space. Only way to be sure".

**Comment by eerhardt** [comment]
When I spoke with @mconnew it seemed that this issue would take a lot of work, and is pretty low on the priority list.

With #35547, we have addressed the size concern with having to root `XmlSerializer.GenerateSerializer`. I'm going to remove the `linkable-framework` label, as this doesn't issue no longer needs to be tracked in the linkablity work.
